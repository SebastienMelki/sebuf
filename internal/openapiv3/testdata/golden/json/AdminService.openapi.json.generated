{"components":{"schemas":{"FieldViolation":{"description":"FieldViolation describes a single validation error for a specific field.","properties":{"description":{"description":"Human-readable description of the validation violation (e.g., 'must be a valid email address', 'required field missing')","type":"string"},"field":{"description":"The field path that failed validation (e.g., 'user.email' for nested fields). For header validation, this will be the header name (e.g., 'X-API-Key')","type":"string"}},"required":["field","description"],"type":"object"},"Request":{"description":"Generic request message","properties":{"data":{"description":"Request data","type":"string"},"id":{"description":"Request ID","type":"string"}},"type":"object"},"Response":{"description":"Generic response message","properties":{"data":{"description":"Response data","type":"string"},"message":{"description":"Response message","type":"string"},"success":{"description":"Success indicator","type":"boolean"}},"type":"object"},"ValidationError":{"description":"ValidationError is returned when request validation fails. It contains a list of field violations describing what went wrong.","properties":{"violations":{"description":"List of validation violations","items":{"$ref":"#/components/schemas/FieldViolation"},"type":"array"}},"required":["violations"],"type":"object"}}},"info":{"title":"AdminService API","version":"1.0.0"},"openapi":"3.1.0","paths":{"/api/v1/admin/stats":{"post":{"description":"Get system stats (admin only)","operationId":"GetSystemStats","parameters":[{"description":"Admin authentication token","in":"header","name":"X-Admin-Token","required":true,"schema":{"format":"uuid","type":"string"}},{"description":"Admin role level","in":"header","name":"X-Admin-Role","required":true,"schema":{"type":"string"}}],"requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/Request"}}},"required":true},"responses":{"200":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/Response"}}},"description":"Successful response"},"400":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/ValidationError"}}},"description":"Validation error"},"default":{"content":{"application/json":{"schema":{"properties":{"code":{"type":"integer"},"error":{"type":"string"}},"type":"object"}}},"description":"Error response"}},"summary":"GetSystemStats","tags":["AdminService"]}},"/api/v1/admin/users/delete":{"post":{"description":"Delete user (admin only)","operationId":"DeleteUser","parameters":[{"description":"Admin role level","in":"header","name":"X-Admin-Role","required":true,"schema":{"type":"string"}},{"description":"Admin authentication token","in":"header","name":"X-Admin-Token","required":true,"schema":{"format":"uuid","type":"string"}},{"description":"Confirmation token for destructive operations","in":"header","name":"X-Confirmation-Token","required":true,"schema":{"type":"string"}}],"requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/Request"}}},"required":true},"responses":{"200":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/Response"}}},"description":"Successful response"},"400":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/ValidationError"}}},"description":"Validation error"},"default":{"content":{"application/json":{"schema":{"properties":{"code":{"type":"integer"},"error":{"type":"string"}},"type":"object"}}},"description":"Error response"}},"summary":"DeleteUser","tags":["AdminService"]}},"/api/v1/admin/users/list":{"post":{"description":"List all users (admin only)","operationId":"ListUsers","parameters":[{"description":"Admin authentication token","in":"header","name":"X-Admin-Token","required":true,"schema":{"format":"uuid","type":"string"}},{"description":"Admin role level","in":"header","name":"X-Admin-Role","required":true,"schema":{"type":"string"}}],"requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/Request"}}},"required":true},"responses":{"200":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/Response"}}},"description":"Successful response"},"400":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/ValidationError"}}},"description":"Validation error"},"default":{"content":{"application/json":{"schema":{"properties":{"code":{"type":"integer"},"error":{"type":"string"}},"type":"object"}}},"description":"Error response"}},"summary":"ListUsers","tags":["AdminService"]}}}}