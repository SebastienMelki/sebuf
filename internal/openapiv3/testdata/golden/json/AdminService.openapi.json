{"components":{"schemas":{"Request":{"description":"Generic request message","properties":{"data":{"description":"Request data","type":"string"},"id":{"description":"Request ID","type":"string"}},"type":"object"},"Response":{"description":"Generic response message","properties":{"data":{"description":"Response data","type":"string"},"message":{"description":"Response message","type":"string"},"success":{"description":"Success indicator","type":"boolean"}},"type":"object"},"User":{"description":"User message used by multiple services","properties":{"email":{"description":"User email","type":"string"},"id":{"description":"User ID","type":"string"},"name":{"description":"User name","type":"string"},"role":{"description":"User role","type":"string"}},"type":"object"}}},"info":{"title":"AdminService API","version":"1.0.0"},"openapi":"3.1.0","paths":{"/api/v1/admin/stats":{"post":{"description":"Get system stats (admin only)","operationId":"GetSystemStats","parameters":[{"description":"Admin authentication token","in":"header","name":"X-Admin-Token","required":true,"schema":{"format":"uuid","type":"string"}},{"description":"Admin role level","in":"header","name":"X-Admin-Role","required":true,"schema":{"type":"string"}}],"requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/Request"}}},"required":true},"responses":{"200":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/Response"}}},"description":"Successful response"},"default":{"content":{"application/json":{"schema":{"properties":{"code":{"type":"integer"},"error":{"type":"string"}},"type":"object"}}},"description":"Error response"}},"summary":"GetSystemStats","tags":["AdminService"]}},"/api/v1/admin/users/delete":{"post":{"description":"Delete user (admin only)","operationId":"DeleteUser","parameters":[{"description":"Admin role level","in":"header","name":"X-Admin-Role","required":true,"schema":{"type":"string"}},{"description":"Admin authentication token","in":"header","name":"X-Admin-Token","required":true,"schema":{"format":"uuid","type":"string"}},{"description":"Confirmation token for destructive operations","in":"header","name":"X-Confirmation-Token","required":true,"schema":{"type":"string"}}],"requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/Request"}}},"required":true},"responses":{"200":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/Response"}}},"description":"Successful response"},"default":{"content":{"application/json":{"schema":{"properties":{"code":{"type":"integer"},"error":{"type":"string"}},"type":"object"}}},"description":"Error response"}},"summary":"DeleteUser","tags":["AdminService"]}},"/api/v1/admin/users/list":{"post":{"description":"List all users (admin only)","operationId":"ListUsers","parameters":[{"description":"Admin authentication token","in":"header","name":"X-Admin-Token","required":true,"schema":{"format":"uuid","type":"string"}},{"description":"Admin role level","in":"header","name":"X-Admin-Role","required":true,"schema":{"type":"string"}}],"requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/Request"}}},"required":true},"responses":{"200":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/Response"}}},"description":"Successful response"},"default":{"content":{"application/json":{"schema":{"properties":{"code":{"type":"integer"},"error":{"type":"string"}},"type":"object"}}},"description":"Error response"}},"summary":"ListUsers","tags":["AdminService"]}}}}